lib/modules/manager/swift/artifacts.ts

```diff
@@ -0,0 +1,104 @@
+import { TEMPORARY_ERROR } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+import {
+  getSiblingFileName,
+  localPathExists,
+  readLocalFile,
+  writeLocalFile,
+} from '../../../util/fs';
+import type {
+  UpdateArtifact,
+  UpdateArtifactsConfig,
+  UpdateArtifactsResult,
+} from '../types';
+import { extractSwiftToolsVersion } from './util';
+
+async function swiftPackageResolve(
+  packageFile: string,
+  config: UpdateArtifactsConfig
+): Promise<void> {
+  const cmd = `swift package resolve`;
+  const execOptions: ExecOptions = {
+    cwdFile: packageFile,
+    docker: {},
+    toolConstraints: [
+      {
+        toolName: 'swift',
+        constraint: config.constraints?.swift,
```

Comment:
```suggestion
        constraint: config.constraints?.swift ?? extractedVersion,
```

use extracted version if found and no constraints are configured.